src/main/java/uk/ac/ebi/eva/commons/core/models/AbstractVariantSourceEntry.java

```diff
@@ -0,0 +1,193 @@
+/*
+ * Copyright 2017 EMBL - European Bioinformatics Institute
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package uk.ac.ebi.eva.commons.core.models;
+
+import java.util.Arrays;
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+/**
+ * Entry that associates a variant and a file in a variant archive. It contains
+ * information related to samples, statistics and specifics of the file format.
+ */
+public abstract class AbstractVariantSourceEntry implements IVariantSourceEntry {
+
+    public static final String DEFAULT_COHORT = "ALL";
+
+    /**
+     * Unique identifier of the archived file.
+     */
+    private String fileId;
+
+    /**
+     * Unique identifier of the study containing the archived file.
+     */
+    private String studyId;
+
+    /**
+     * Alternate alleles that appear along with a variant alternate.
+     */
+    private String[] secondaryAlternates;
+
+    /**
+     * Fields stored for each sample.
+     */
+    private String format;
+
+    /**
+     * Statistics of the genomic variation, such as its alleles/genotypes count
+     * or its minimum allele frequency, grouped by cohort name.
+     */
+    private Map<String, VariantStats> cohortStats;
+
+    /**
+     * Optional attributes that probably depend on the format of the file the
```

Comment:
This should be noted for a full renaming then as there are a lot files/sources concepts mixed.